src/test/librbd/test_librbd.cc

```diff
@@ -210,10 +210,38 @@ class TestLibRBD : public ::testing::Test {
     ASSERT_NE("", m_pool_name = create_pool());
   }
 
-  bool is_skip_partial_discard_enabled() {
+  bool is_skip_partial_discard_enabled(rbd_image_t image) {
     std::string value;
     EXPECT_EQ(0, _rados.conf_get("rbd_skip_partial_discard", value));
-    return value == "true";
+
+    if (value == "true") {
+      int r = rbd_lock_acquire(image, RBD_LOCK_MODE_EXCLUSIVE);
```

Comment:
Nit: why do you need to acquire the lock to determine the feature bits?

Reply:
Currently, when rwl is enabled, the discard granularity is set to 0. The dirty_cache feature bit is set after getting an exclusive lock. When an rbd image is opened, the dirty_cache feature may not be set.

Reply:
Perhaps just send a flush request?